lib/services/file.js

```diff
@@ -47,6 +48,24 @@ class File extends Service {
         new winston.transports.File(conf)
       ]
     });
+
+    try {
+      if (config.filename) {
+        this.isWritableSync(config.filename);
+      }
+    } catch (error) {
+      if (error.code === 'EPERM') {
+        throw new kerror.PreconditionError(`[kuzzle-plugin-logger] EPERM, cannot open ${config.filename}`);
+      }
+
+      throw error;
+    }
+  }
```

Comment:
This format looks weird IMO, maybe something like this:
```suggestion
    if (config.filename) {
      try {
          this.isWritableSync(config.filename);
      } catch (error) {
        if (error.code === 'EPERM') {
          throw new kerror.PreconditionError(`[kuzzle-plugin-logger] EPERM, cannot open ${config.filename}`);
        }
        throw error;
      }
    }
  }
```

Reply:
You are right, guard should be used first over try/catch statement